src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputUnionTypeConverter.cs

```diff
@@ -44,7 +47,7 @@ public static InputUnionType CreateInputUnionType(ref Utf8JsonReader reader, str
                 throw new JsonException("Union must have variant types.");
             }
 
-            var unionType = new InputUnionType(name, variantTypes);
+            var unionType = new InputUnionType(name, variantTypes) { Decorators = decorators ?? Array.Empty<InputDecoratorInfo>() };
```

Comment:
FYI since we have dotnet 8 here as well, we could also have
```suggestion
            var unionType = new InputUnionType(name, variantTypes) { Decorators = decorators ?? [] };
```

Reply:
but please do not consider this as a blocker, we do not need to change this, just keep in mind in the future we should use collection expression.

Reply:
Thanks.  This feature need c# 12 or higher, but autorest.csharp use c# 11, so we cannot use this.